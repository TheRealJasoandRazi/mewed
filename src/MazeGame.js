import { AppBar, Toolbar } from '@mui/material';
import * as React from 'react';
import {
    Stack,
    Container,
    IconButton,
    Box,
    ImageList,
    ImageListItemBar,
    ImageListItem,
    TextField,
    ThemeProvider,
    MenuItem,
    Menu,
    Typography,
    Link,
  } from "@mui/material";
  import image1 from './images/image1.png';
  import image2 from './images/image2.png';
  import image4 from './images/image4.png';
  import image5 from './images/image5.png';

  import video1 from './videos/video1.gif';
  import video2 from './videos/video2.gif';
  import video3 from './videos/video3.gif';

  import List from "./List";

  const itemData = [
    {
      img: image1,
      desc: 'biggest size maze as a 30x30',
    },
    {
      img: image2,
      desc: 'After the computer found the solution',
    },
    {
      img: image4,
      desc: 'the medium sized maze at 20x20',
    },
    {
      img: image5,
      desc: 'the second colour pallet option',
    }];

  const videoData = [
    {
      img: video1,
      desc: 'the computer finding the solution',
    },{
      img: video2,
      desc: 'changing colour pallets',
    },
    {
      img: video3,
      desc: 'changing the sizes of maze and layout',
    }
  ];

  function MazeGame() {
    return (
      <Container
      fixed
      sx={{
        bgcolor: "transparent",
        marginTop: '10vh', 
        marginBottom: '10vh',
      }}
    >
      <Box sx={{ pt: "10px" }}>
        <h1>Maze Game</h1>
        <Stack spacing={2} sx={{ mb: "10px", borderBottom: "2px solid white" }}>
          <Typography variant="h6">Introduction:</Typography>
          <Typography>
            Using the SDL library with C++ i was able to design and develop a maze game. This maze game allows you to regenerate
            a new maze with just a click of a button. There are 3 maze sizes to choose from, a 12x12 a 20x20 and a 30x30 maze size.
            In addition you change the colour pallet from a blue/white layout to a green/black one. Lastly, if you get stuck you can 
            click the "solve" button and the computer will show you how to complete the maze! If you want to look at how the program 
            looks like you can check out the screenshots and gifs below.
          </Typography>
          <Link href="https://github.com/TheRealJasoandRazi/mewed" target="_blank" rel="noopener noreferrer">
            Source Code Here
          </Link>
        </Stack>
        <Box mt={2}>
          <Typography variant="h6">Screenshots</Typography>
            <List items={itemData}/>
          <Typography variant="h6">Gifs</Typography>
            <List items={videoData} />
        </Box>
      </Box>
    </Container>
    );
  };

  export default MazeGame;