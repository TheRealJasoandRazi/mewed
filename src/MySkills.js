import * as React from 'react';
import {
  Typography,
  Paper,
  Box,
} from "@mui/material";
import PropTypes from 'prop-types';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';

import Grow from '@mui/material/Grow';

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
};

function a11yProps(index) {
  return {
    id: `vertical-tab-${index}`,
    'aria-controls': `vertical-tabpanel-${index}`,
  };
}

const icon = (
  <Paper sx={{ m: 1, width: 100, height: 100 }} elevation={4}>
    <svg>
      <Box
        component="polygon"
        points="0,100 50,00, 100,100"
        sx={{
          fill: (theme) => theme.palette.common.white,
          stroke: (theme) => theme.palette.divider,
          strokeWidth: 1,
        }}
      />
    </svg>
  </Paper>
);

function MySkills() {
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <>
      <h1
        style={{textAlign: 'center'}}
      >
        My Skills</h1>
      <Box
        sx={{ flexGrow: 1, bgcolor: 'background.paper', display: 'flex', width: '90%', margin: '0 auto' }}
      >
        <Tabs
          orientation="vertical"
          variant="scrollable"
          value={value}
          onChange={handleChange}
          aria-label="Vertical tabs example"
          sx={{ borderRight: 1, borderColor: 'divider' }}
        >
          <Tab label="Web Development" {...a11yProps(0)} />
          <Tab label="Programming" {...a11yProps(1)} />
          <Tab label="Cloud" {...a11yProps(2)} />
          <Tab label="DevOps" {...a11yProps(3)} />
          <Tab label="DataBases" {...a11yProps(4)} />
        </Tabs>
        <TabPanel value={value} index={0}>
          <Grow in={true}>
            <Typography sx={{ color: 'black' }}>JavaScript - This website is made with React.js!</Typography>
          </Grow>
          <Grow in={true} timeout={1000}>
            <Typography sx={{ color: 'black' }}>PHP</Typography>
          </Grow>
          <Grow in={true} timeout={2000}>
            <Typography sx={{ color: 'black' }}>HTML/CSS</Typography>
          </Grow>
        </TabPanel>
        <TabPanel value={value} index={1}>
          <Grow in={true}>
            <Typography sx={{ color: 'black' }}>C++ - Maze Game from Personal Projects</Typography>
          </Grow>
          <Grow in={true} timeout={1000}>
            <Typography sx={{ color: 'black' }}>C#</Typography>
          </Grow>
          <Grow in={true} timeout={2000}>
            <Typography sx={{ color: 'black' }}>Python</Typography>
          </Grow>
        </TabPanel>
        <TabPanel value={value} index={2}>
          <Grow in={true}>
            <Typography sx={{ color: 'black' }}>AWS</Typography>
          </Grow>
          <Grow in={true} timeout={1000}>
            <Typography sx={{ color: 'black' }}>I am currently pursuing an AWS Associate Solutions Architect certification</Typography>
          </Grow>
        </TabPanel>
        <TabPanel value={value} index={3}>
          <Grow in={true}>
            <Typography sx={{ color: 'black' }}>JenKins</Typography>
          </Grow>
          <Grow in={true} timeout={1000}>
            <Typography sx={{ color: 'black' }}>Selenium</Typography>
          </Grow>
          <Grow in={true} timeout={2000}>
            <Typography sx={{ color: 'black' }}>Prometheus</Typography>
          </Grow>
          <Grow in={true} timeout={3000}>
            <Typography sx={{ color: 'black' }}>Grafana</Typography>
          </Grow>
        </TabPanel>
        <TabPanel value={value} index={4}>
          <Grow in={true}>
            <Typography sx={{ color: 'black' }}>SQL - Used in university projects</Typography>
          </Grow>
        </TabPanel>
      </Box>
    </>
  );
}

export default MySkills;
