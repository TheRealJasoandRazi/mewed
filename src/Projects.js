import { Flex, Container, Box, Stack, Typography, Card, Divider, CardMedia, Grid } from "@mui/material";
import image from './images/image5.png';
import DevOps from './images/DevOps.png';
import TradingPlatform from './images/marketplace.png';

const commonCardStyle = {
  bgcolor: 'cyan',
}

const MazeGameCard = (
  <Card sx={commonCardStyle}>
    <Typography sx={{ fontSize: 20 }} color="text.primary" gutterBottom>
      Maze Game
    </Typography>
    <Divider />
    <Box
    style={{
      height: '40vh',
    }}>
      <CardMedia
          component="img"
          alt="Project Image"
          src={image}
          style={{
            objectFit: "cover",
            width: "100%",
            height: "100%",
          }}
        />
    </Box>
  </Card>
);

const MockPlatformCard = (
  <Card sx={commonCardStyle}>
    <Typography sx={{ fontSize: 20 }} color="text.primary" gutterBottom>
      Mock Trading Platform
    </Typography>
    <Divider />
    <Box
    style={{
      height: '40vh',
    }}>
      <CardMedia
        component="img"
        alt="Project Image"
        src={TradingPlatform}
        style={{
          objectFit: "cover",
          width: "100%",
          height: "100%",
        }}
      />
    </Box>
  </Card>
);

function Projects() {
  return (
    <Container
      fixed
      sx={{
        bgcolor: "transparent",
        marginTop: '10vh',
        marginBottom: '10vh',
        textAlign: 'center',
      }}
    >
        <h1>Projects</h1>
        <Stack direction="row" spacing={5} sx={{ 
            justifyContent: 'flex-end',
            padding: '1vh' 
        }}>
          <Box sx={{ flex: 1, textAlign: 'center' }}>
            <a href="/MazeGame" style={{ textDecoration: "none" }}>
              {MazeGameCard}
            </a>
          </Box>
          <Box sx={{ flex: 1, textAlign: 'center'}}>
            <a href="/MockTradingPlatform" style={{ textDecoration: "none" }}>
              {MockPlatformCard}
            </a>
          </Box>
      </Stack>
    </Container>
  );
}

export default Projects;
