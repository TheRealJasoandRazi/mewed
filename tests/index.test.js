import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Projects from '../src/Projects.js';
import MazeGame from '../src/MazeGame.js';
import MySkills from '../src/MySkills.js';


describe('components render correctly', () => {
    test('projects', () => {
        const component = renderer.create(<Projects />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
