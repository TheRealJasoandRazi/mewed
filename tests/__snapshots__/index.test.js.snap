// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1a5yo9g-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;padding-top:1%;}.css-1a5yo9g-MuiStack-root>:not(style):not(style){margin:0;}.css-1a5yo9g-MuiStack-root>:not(style)~:not(style){margin-top:16px;}",
      }
    }
    data-emotion="css 1a5yo9g-MuiStack-root"
  />,
  <div
    className="MuiStack-root css-1a5yo9g-MuiStack-root"
  >
    <div
      style={
        Object {
          "alignContent": "center",
          "background": "#ff6d6d",
          "borderRadius": 8,
          "height": "33%",
          "textAlign": "center",
          "transform": "none",
          "width": "50%",
        }
      }
    >
      <h1>
        My Projects
      </h1>
    </div>
    <div
      style={
        Object {
          "alignContent": "center",
          "background": "#ff6d6d",
          "borderRadius": 8,
          "height": "33%",
          "textAlign": "center",
          "transform": "none",
          "width": "50%",
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1veoth8-MuiCard-root{overflow:hidden;background-color:cyan;height:300;}",
          }
        }
        data-emotion="css 1veoth8-MuiCard-root"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-vqbcwq-MuiPaper-root-MuiCard-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);overflow:hidden;background-color:cyan;height:300;}",
          }
        }
        data-emotion="css vqbcwq-MuiPaper-root-MuiCard-root"
      />
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vqbcwq-MuiPaper-root-MuiCard-root"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-vfu6h6-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;margin-bottom:0.35em;color:rgba(0, 0, 0, 0.87);font-size:20px;}",
            }
          }
          data-emotion="css vfu6h6-MuiTypography-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-vfu6h6-MuiTypography-root"
        >
          Maze Game
        </p>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-9mgopn-MuiDivider-root{margin:0;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;border-width:0;border-style:solid;border-color:rgba(0, 0, 0, 0.12);border-bottom-width:thin;}",
            }
          }
          data-emotion="css 9mgopn-MuiDivider-root"
        />
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          className="MuiBox-root css-0"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-o69gx8-MuiCardMedia-root{display:block;-webkit-background-size:cover;background-size:cover;background-repeat:no-repeat;-webkit-background-position:center;background-position:center;width:100%;object-fit:cover;}",
              }
            }
            data-emotion="css o69gx8-MuiCardMedia-root"
          />
          <img
            alt="Project Image"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            src=""
            style={
              Object {
                "height": "100%",
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignContent": "center",
          "background": "#ff6d6d",
          "borderRadius": 8,
          "height": "33%",
          "textAlign": "center",
          "transform": "none",
          "width": "50%",
        }
      }
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vqbcwq-MuiPaper-root-MuiCard-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-vfu6h6-MuiTypography-root"
        >
          Mock Trading Platform
        </p>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <img
          alt="Project Image"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          src=""
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>,
]
`;
